File: prog-back/internal/models/reservation.go
package models

import "time"

type Reservation struct {
	ID                 uint      `gorm:"primaryKey" json:"id"`
	RoomID             uint      `gorm:"index;not null" json:"room_id"`
	Room               *Room     `gorm:"foreignKey:RoomID" json:"room,omitempty"`
	UserID             uint      `gorm:"index;not null" json:"user_id"`
	User               *User     `gorm:"foreignKey:UserID" json:"user,omitempty"`
	ClassID            *uint     `gorm:"index" json:"class_id,omitempty"`
	Class              *Class    `gorm:"foreignKey:ClassID" json:"class,omitempty"`
	StartTime          time.Time `gorm:"not null;index" json:"start_time"`
	EndTime            time.Time `gorm:"not null;index" json:"end_time"`
	Purpose            string    `json:"purpose"`
	EstimatedAttendees int       `json:"estimated_attendees"`
	Status             string    `gorm:"not null;default:'ACTIVE'" json:"status"` // ACTIVE|CANCELLED
	CreatedAt          time.Time `json:"created_at"`
	UpdatedAt          time.Time `json:"updated_at"`
}

File: prog-front/src/lib/types/reservation.ts
import { Room } from "./room";
import { User } from "./user";
import { Class } from "./class";

// Tipos de reservas
export interface Reservation {
  id: number;
  room_id: number;
  room?: Room;
  user_id: number;
  user?: User;
  class_id?: number;
  class?: Class;
  start_time: string; // ISO 8601
  end_time: string; // ISO 8601
  purpose: string;
  estimated_attendees: number;
  status: "ACTIVE" | "CANCELLED";
  created_at: string;
  updated_at: string;
}

export interface CreateReservationData {
  room_id: number;
  class_id?: number;
  start_time: string;
  end_time: string;
  purpose: string;
  estimated_attendees: number;
}

export interface ReservationFilters {
  room_id?: number;
  building_id?: number;
  class_id?: number;
  user_id?: number;
  status?: "ACTIVE" | "CANCELLED";
  start_date?: string;
  end_date?: string;
}
